src/jit/codegenarmarch.cpp

```diff
@@ -1711,38 +1711,24 @@ void CodeGen::genCodeForIndexAddr(GenTreeIndexAddr* node)
         genJumpToThrowHlpBlk(genJumpKindForOper(GT_GE, CK_UNSIGNED), SCK_RNGCHK_FAIL, node->gtIndRngFailBB);
     }
 
-    // Compute the address of the array element.
-    switch (node->gtElemSize)
+    // Can we use a ScaledAdd instruction?
+    //
+    if (isPow2(node->gtElemSize) && (node->gtElemSize <= 32768))
```

Comment:
Why the magic number 32768?  Looks like arm64 supports shifts of 0-31/63.  Arm32 at first glance should be similar.

Reply:
I could increase this to a larger power of two, but would still need some upper bound check, 
so I picked this value as a reasonable limit check.